#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.0759483444156471 -0.974052165927407 -0.2131999695887626 2.220262238146504
  position -3.0198072526255717 2.1875038570451935 -2.538665596492556
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
  floorTileSize 1 1
  wallThickness 0.05
  wallHeight 0.5
}
Robot {
  translation 1.93004 0.04 0.358274
  rotation 0 1 0 -1.4956853071795866
  children [
    Group {
      children [
        Compass {
        }
      ]
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.1 0.059 0.215
            0.1 0.059 0.215
            0.135 0.059 0.185
            0.135 0.059 -0.095
            0.08 0.059 -0.11
            -0.08 0.059 -0.11
            -0.135 0.059 -0.095
            -0.135 0.059 0.185
            -0.1 0.059 0.215
            -0.1 0.234 0.215
            0.1 0.234 0.215
            0.135 0.234 0.185
            0.135 0.234 -0.095
            0.08 0.234 -0.11
            -0.08 0.234 -0.11
            -0.135 0.234 -0.095
            -0.135 0.234 0.185
            -0.1 0.234 0.215
          ]
        }
        coordIndex [
          8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 9, 10, 11, 12, 13, 14, 15, 16, 17, -1, 0, 1, 10, 9, -1, 1, 2, 11, 10, -1, 2, 3, 12, 11, -1, 3, 4, 13, 12, -1, 4, 5, 14, 13, -1, 5, 6, 15, 14, -1, 6, 7, 16, 15, -1, 7, 8, 17, 16, -1
        ]
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.636415 0.655556 0.655556
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0 0.234 -0.174, -0.056 0.234 -0.166, -0.107 0.234 -0.145, -0.155 0.234 -0.112, -0.19 0.234 -0.064, -0.19 0.234 0.074, -0.16 0.234 0.096, -0.16 0.234 0.151, -0.155 0.234 0.2, -0.107 0.234 0.236, -0.056 0.234 0.256, 0 0.234 0.264, 0.056 0.234 0.256, 0.107 0.234 0.236, 0.155 0.234 0.2, 0.16 0.234 0.151, 0.16 0.234 0.096, 0.19 0.234 0.074, 0.19 0.234 -0.064, 0.155 0.234 -0.112, 0.107 0.234 -0.145, 0.056 0.234 -0.166, 0 0.234 -0.174, 0 0.245 -0.174, -0.056 0.245 -0.166, -0.107 0.245 -0.145, -0.155 0.245 -0.112, -0.19 0.245 -0.064, -0.19 0.245 0.074, -0.16 0.245 0.096, -0.16 0.245 0.151, -0.155 0.245 0.2, -0.107 0.245 0.236, -0.056 0.245 0.256, 0 0.245 0.264, 0.056 0.245 0.256, 0.107 0.245 0.236, 0.155 0.245 0.2, 0.16 0.245 0.151, 0.16 0.245 0.096, 0.19 0.245 0.074, 0.19 0.245 -0.064, 0.155 0.245 -0.112, 0.107 0.245 -0.145, 0.056 0.245 -0.166, 0 0.245 -0.174
          ]
        }
        coordIndex [
          22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, -1, 0, 1, 24, 23, -1, 1, 2, 25, 24, -1, 2, 3, 26, 25, -1, 3, 4, 27, 26, -1, 4, 5, 28, 27, -1, 5, 6, 29, 28, -1, 6, 7, 30, 29, -1, 7, 8, 31, 30, -1, 8, 9, 32, 31, -1, 9, 10, 33, 32, -1, 10, 11, 34, 33, -1, 11, 12, 35, 34, -1, 12, 13, 36, 35, -1, 13, 14, 37, 36, -1, 14, 15, 38, 37, -1, 15, 16, 39, 38, -1, 16, 17, 40, 39, -1, 17, 18, 41, 40, -1, 18, 19, 42, 41, -1, 19, 20, 43, 42, -1, 20, 21, 44, 43, -1, 21, 22, 45, 44, -1
        ]
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 181.09439999999756
        axis -1 0 0
        anchor 0 0.0825 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.1603 0.0825 0
        rotation 1 0 0 0.07562180103085944
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.0037037 0.0037037 0.0037037
                  roughness 0.3
                  metalness 0
                }
                geometry Cylinder {
                  height 0.037
                  radius 0.0825
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 187.14367999999743
        axis -1 0 0
        anchor 0 0.0825 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.1603 0.0825 0
        rotation 1 0 0 2.2745654453901634
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.136 0.184 -0.046
            0.136 0.184 -0.096
            0.101 0.184 -0.134
            0.054 0.184 -0.159
            0 0.184 -0.168
            -0.054 0.184 -0.159
            -0.101 0.184 -0.134
            -0.136 0.184 -0.096
            -0.136 0.184 -0.046
            0.136 0.184 -0.043
            0.136 0.234 -0.046
            0.136 0.234 -0.096
            0.101 0.234 -0.134
            0.054 0.234 -0.159
            0 0.234 -0.168
            -0.054 0.234 -0.159
            -0.101 0.234 -0.134
            -0.136 0.234 -0.096
            -0.136 0.234 -0.046
            0.136 0.234 -0.043
          ]
        }
        coordIndex [
          9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, -1, 0, 1, 11, 10, -1, 1, 2, 12, 11, -1, 2, 3, 13, 12, -1, 3, 4, 14, 13, -1, 4, 5, 15, 14, -1, 5, 6, 16, 15, -1, 6, 7, 17, 16, -1, 7, 8, 18, 17, -1, 8, 9, 19, 18, -1, 9, 0, 10, 19, -1
        ]
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.3
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            -0.136 0.184 0.135
            -0.136 0.184 0.185
            -0.101 0.184 0.223
            -0.054 0.184 0.248
            0 0.184 0.257
            0.054 0.184 0.248
            0.101 0.184 0.223
            0.136 0.184 0.185
            0.136 0.184 0.135
            -0.136 0.184 0.135
            -0.136 0.234 0.135
            -0.136 0.234 0.185
            -0.101 0.234 0.223
            -0.054 0.234 0.248
            0 0.234 0.257
            0.054 0.234 0.248
            0.101 0.234 0.223
            0.136 0.234 0.185
            0.136 0.234 0.135
            -0.136 0.234 0.135
          ]
        }
        coordIndex [
          9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, -1, 0, 1, 11, 10, -1, 1, 2, 12, 11, -1, 2, 3, 13, 12, -1, 3, 4, 14, 13, -1, 4, 5, 15, 14, -1, 5, 6, 16, 15, -1, 6, 7, 17, 16, -1, 7, 8, 18, 17, -1, 8, 9, 19, 18, -1
        ]
      }
    }
    Transform {
      translation 0 0.0325 0.2147
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0037037 0.0037037 0.0037037
            roughness 0.3
            metalness 0
          }
          geometry Cylinder {
            height 0.024
            radius 0.0325
          }
        }
      ]
    }
    DistanceSensor {
      translation -0.027 0.209 -0.164
      rotation 0 1 0 1.745
      children [
        DEF SONAR Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 1 0.0517103
                roughness 0.3
                metalness 0
              }
              geometry Cylinder {
                height 0.002
                radius 0.0175
              }
            }
          ]
        }
      ]
      name "ds0"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        1 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.077 0.209 -0.147
      rotation 0 1 0 2.094
      children [
        USE SONAR
      ]
      name "ds1"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.118 0.209 -0.116
      rotation 0 1 0 2.443
      children [
        USE SONAR
      ]
      name "ds2"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.136 0.209 -0.071
      rotation 0 1 0 3.14
      children [
        USE SONAR
      ]
      name "ds3"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.136 0.209 0.16
      rotation 0 1 0 -3.14
      children [
        USE SONAR
      ]
      name "ds4"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.118 0.209 0.205
      rotation 0 1 0 -2.443
      children [
        USE SONAR
      ]
      name "ds5"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.077 0.209 0.236
      rotation 0 1 0 -2.094
      children [
        USE SONAR
      ]
      name "ds6"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation -0.027 0.209 0.253
      rotation 0 1 0 -1.745
      children [
        USE SONAR
      ]
      name "ds7"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.027 0.209 0.253
      rotation 0 1 0 -1.396
      children [
        USE SONAR
      ]
      name "ds8"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.077 0.209 0.236
      rotation 0 1 0 -1.047
      children [
        USE SONAR
      ]
      name "ds9"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.118 0.209 0.205
      rotation 0 1 0 -0.698
      children [
        USE SONAR
      ]
      name "ds10"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.136 0.209 0.16
      children [
        USE SONAR
      ]
      name "ds11"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.136 0.209 -0.071
      children [
        USE SONAR
      ]
      name "ds12"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.118 0.209 -0.116
      rotation 0 1 0 0.698
      children [
        USE SONAR
      ]
      name "ds13"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.077 0.209 -0.147
      rotation 0 1 0 1.047
      children [
        USE SONAR
      ]
      name "ds14"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        0.4 0 0.1
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.027 0.209 -0.164
      rotation 0 1 0 1.396
      children [
        USE SONAR
      ]
      name "ds15"
      lookupTable [
        0 1024 0
        0.2 1024 0.1
        1 0 0.1
      ]
      type "infra-red"
    }
  ]
  name "Pioneer 2"
  description "designed by Activmedia Robotics"
  boundingObject Transform {
    translation 0 0.13 0.0447
    children [
      Cylinder {
        height 0.24
        radius 0.219
      }
    ]
  }
  controller "wall_follower"
}
Solid {
  translation -1 0.25 1.42
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.1 0.5 2
      }
    }
  ]
  boundingObject Box {
    size 0.1 0.5 2
  }
}
Solid {
  translation 1.55973 0.25 -1.33589
  rotation 0 1 0 1.8326
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.1 0.5 2
      }
    }
  ]
  name "solid(1)"
  boundingObject Box {
    size 0.1 0.5 2
  }
}
